subsys/logging/log_core.c

```diff
@@ -15,6 +15,10 @@
 #include <atomic.h>
 #include <ctype.h>
 
+#if defined(CONFIG_ARM)
+#include <arch/arm/cortex_m/memory_map.h>
```

Comment:
I think the defines in this file are not used; i would hope we could get rid of this header, so using it here makes it harder.